pkg/kube/cr_manager.go

```diff
@@ -148,54 +149,56 @@ func NewCRManager(clientset kubernetes.Interface, clientsetext extapi.Apiextensi
 	}
 }
 
-func (m *crManager) Init(ctx context.Context) (err error) {
-	err = m.createOrUpdateCRD(ctx, &aquasecurityv1alpha1.VulnerabilityReportsCRD)
+func (m *crManager) Init(ctx context.Context, initOptions *InitOptions, outWriter io.Writer) (err error) {
+	klog.V(3).Infof("In DryRun mode=%v", initOptions.DryRun)
```

Comment:
I think we could simplify this logic and resolve --dry-run mode in one place instead of spreading this logic in so many places.

I'd actually start with checking if dry-run mode is `client` and immediately short circuit the `Init` method with print statements showing which objects will be created without sending any requests to Kubernetes API.

Then we have to handle `none` and `server` modes, but this I'd even resolve in the construction and store CreateOptions and UpeateOptions structs as crManager fields, i.e.:

```go
type crManager struct {
	clientset    kubernetes.Interface
	clientsetext extapi.ApiextensionsV1beta1Interface
	createOptions meta.CreateOptions
	updateOptions meta.UpdateOptions
}

// NewCRManager constructs a CRManager with the given Kubernetes interface.
func NewCRManager(opt InitOptions, clientset kubernetes.Interface, clientsetext extapi.ApiextensionsV1beta1Interface) CRManager {
	return &crManager{
		clientset:    clientset,
		clientsetext: clientsetext,
		createOptions: meta.CreateOptions{
			// Set DryRun property based on the opt passed as constructor arg
			DryRun: []string{},               // if --dry-run=none
			DryRun: []string{meta.DryRunAll}, // if --dry-run=server
		},
		updateOptions: meta.UpdateOptions{
			DryRun: []string{},
			DryRun: []string{meta.DryRunAll},
		},
	}
}
```

WDYT?

Reply:
Now that I have been through the process and know how it works... then I would be in favour of simplifying the logic. Your approach sounds sensible, so I will refactor accordingly.